src/wmtk/EdgeMeshOperationExecutor.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+#include <wmtk/utils/Logger.hpp>
+#include "EdgeMesh.hpp"
+#include "SimplicialComplex.hpp"
+#include "Tuple.hpp"
+namespace wmtk {
+class EdgeMesh::EdgeMeshOperationExecutor
+{
+public:
+    EdgeMeshOperationExecutor(EdgeMesh& m, const Tuple& operating_tuple, Accessor<long>& hash_acc);
+    void delete_simplices();
+    void update_cell_hash();
+
+    std::array<Accessor<char>, 2> flag_accessors;
+    Accessor<long> ee_accessor;
+    Accessor<long> ev_accessor;
+    Accessor<long> ve_accessor;
+    Accessor<long>& hash_accessor;
+
+    /**
+     * @brief gather all simplices that are deleted in a split
+     *
+     * The deleted simplices are exactly the open star of the edge
+     */
+    static const std::array<std::vector<long>, 2> get_split_simplices_to_delete(
+        const Tuple& tuple,
+        const EdgeMesh& m);
+
+    /**
+     * @brief gather all simplices that are deleted in a collapse
+     *
+     * The deleted simplices are the intersection of the open star of the vertex and the closed star
+     * of the edge. This comes down to one vertex, three edges, and two faces if the edge is on the
+     * interior. On the boundary it is one vertex, two edges, and one face.
+     */
+    static const std::array<std::vector<long>, 2> get_collapse_simplices_to_delete(
+        const Tuple& tuple,
+        const EdgeMesh& m);
+
+    const std::array<long, 2>& incident_vids() const { return m_spine_vids; }
+
+    long operating_edge_id() const { return m_operating_edge_id; }
+
+    Tuple split_edge();
+    Tuple collapse_edge();
+    Tuple split_edge_single_mesh();
```

Comment:
I feel like these `_single_` functions should be private. Correct me if I am wrong.